projects/components/src/common/activity-reporter/banner-activity-reporter.component.ts

```diff
@@ -75,6 +77,14 @@ export class BannerActivityReporterComponent extends ActivityReporter {
         this.successMessage = null;
     }
 
+    /**
+     * To view error and success messages, scroll to the top of the page for visibility
+     */
+    scrollToMessage(): void {
+        const element = document.querySelector('vcd-banner-activity-reporter');
```

Comment:
This is not a safe way, you could have two banners on the page. The correct way is to get a reference to the component's element and query using that.

If you are ever forced to use `document` in Angular, you should always think twice, and if really required, document why.

This method should be private since we should always minimize a component's public API. 

The method name is misleading since it's not scrolling the component itself, it's scrolling components around it so that it comes into view, please rename it like the HTML method, `scrollIntoView`. The comment is also either inaccurate or ambiguous since you are not necessarily scrolling to the top of the page. Scrolling to the top of the page may not bring an element into view, it may need to be scrolled down so that it's visible, don't embed the assumption that this component must be used at the top of a page.

The variable that was created doesn't help code comprehension. it's only used once after it was created and its name is very generic.

```ts
constructor(private elementRef: ElementRef) { }

private scrollIntoView: void {
  this.elementRef.nativeElement.scrollIntoView();
}
```

Reply:
Addressed the comments from above in my new changes.